pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -315,6 +315,18 @@ func (s *FlinkStateMachine) handleClusterStarting(ctx context.Context, applicati
 	}
 	return statusChanged, nil
 }
+func (s *FlinkStateMachine) handleApplicationSavepointingWithCheckpoint(ctx context.Context, application *v1beta1.FlinkApplication) (bool, error) {
+	checkpointPath, err := s.flinkController.FindExternalizedCheckpointForSavepoint(ctx, application, application.Status.DeployHash)
```

Comment:
Cancelling the job would still be required here, correct?

Reply:
We still need to cancel, yes, and maybe I didn't read closely enough but: `handleApplicationSavepointingWithCheckpoint` is called within `handleApplicationSavepointing` which also does not cancel. I figured this area of code was only responsible for ensuring a valid savepoint path had been harvested and stored on `application.Status.SavepointPath`.

I may be wrong in invoking the new function in `handleApplicationSavepointing` since savepoints can be requested directly on the cancel API call.

I'll review again.